src/codemagic/cli/cli_app.py

```diff
@@ -208,10 +226,12 @@ def invoke_cli(cls) -> NoReturn:
                 raise argparse.ArgumentError(args.action, "the following argument is required: action")
             elif cls._action_requires_subcommand(args.action) and not args.action_subcommand:
                 raise argparse.ArgumentError(args.action_subcommand, "the following argument is required: subcommand")
+            elif not cls._validate_args(args):
+                pass  # In case of invalid args validation will raise
```

Comment:
should be made clear that the condition is never `True`
```suggestion
            elif not cls._validate_args(args):
                assert_never()  # In case of invalid args validation will raise
```
https://docs.python.org/3/library/typing.html#typing.assert_never